server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -235,7 +239,7 @@ private static int indexOfKeptCommits(List<? extends IndexCommit> commits, long
      */
     boolean hasUnreferencedCommits() throws IOException {
         final IndexCommit lastCommit = this.lastCommit;
-        if (safeCommit != lastCommit) { // Race condition can happen but harmless
+        if (safeCommit != lastCommit && pendingSnapshots == 0) { // Race condition can happen but harmless
```

Comment:
I don't really follow this logic? safe commit is maintained outside of the real of snapshots? shouldn't these be two different things? also, we now call this method when the translog is synced and thus potentially something changed for the safe commit. That doesn't have much todo with snapshots and the chance of them being released. Maybe we should make release commit return a boolean and use that an indication that there are commits that can be cleaned up?